packages/client/src/pinnedBlocks.ts

```diff
@@ -0,0 +1,58 @@
+import { BlockInfo, PinnedBlocks } from "@polkadot-api/observable-client"
+import {
+  distinctUntilChanged,
+  map,
+  mergeMap,
+  Observable,
+  scan,
+  share,
+} from "rxjs"
+import { shareLatest } from "./utils"
+
+export const getNewBlocks$ = (pinnedBlocks$: Observable<PinnedBlocks>) =>
+  pinnedBlocks$.pipe(
+    scan(
+      ({ reportedBlocks: prevReportedBlocks }, { blocks }) => {
+        const reportedBlocks = new Set<string>(blocks.keys())
+        const newBlocks: BlockInfo[] = []
+
+        if (reportedBlocks.size > prevReportedBlocks.size) {
+          blocks.forEach(({ hash, number, parent }) => {
+            if (!prevReportedBlocks.has(hash)) {
+              newBlocks.push({
+                hash,
+                number,
+                parent,
+              })
+            }
+          })
+        }
+
+        return { reportedBlocks, newBlocks }
+      },
+      {
+        reportedBlocks: new Set<string>(),
+        newBlocks: new Array<BlockInfo>(0),
+      },
+    ),
+    mergeMap(({ newBlocks }) => newBlocks),
```

Comment:
nice! very clever! 🌈 

However, as we discussed offline probably best to move this to the observable-client